app/validation/validator.py

```diff
@@ -729,9 +729,45 @@ def _validate_routing_rule(self, rule, answer_ids_with_group_id, block_or_group)
                     rule["when"], answer_ids_with_group_id, block_or_group["id"]
                 )
             )
+            for when_dict in rule.get("when"):
+                errors.extend(
+                    self._validate_routing_when_dict(
+                        when_dict, answer_ids_with_group_id
+                    )
+                )
+
+        return errors
+
+    def _validate_routing_when_dict(self, when_dict, answer_ids_with_parent_id):
+        errors = []
+        when_values = when_dict.get("values", [])
+        when_value = when_dict.get("value")
+        answer_id = when_dict.get("id", None)
```

Comment:
```suggestion
        answer_id = when_dict.get("id")
```

None is the default return value

Reply:
Done, thanks!